pytorch_lightning/trainer/trainer.py

```diff
@@ -2048,7 +2063,11 @@ def num_gpus(self) -> int:
 
     @property
     def devices(self) -> Optional[Union[List[int], str, int]]:
-        return self._accelerator_connector.devices
+        rank_zero_deprecation(
```

Comment:
can you add a deprecation test?

Reply:
Added in `tests/trainer/test_trainer.py`